js-src/functions/src/main/scala/org/scalalang/utils/OS.scala

```diff
@@ -0,0 +1,55 @@
+package org.scalalang.utils
+
+import org.scalajs.dom.window
+
+/**
+  * Find which OS our client is running
+  */
+sealed trait OS {
+  val navigator: String
+  val label: String
```

Comment:
In order to cut down on the boilerplate, consider declaring `OS` as
```scala
sealed abstract class OS(val navigator: String, val label: String)
```
so that the instances are one-liners:
```scala
case object Windows extends OS("Win", "Windows")
```